pkg/distributor/query.go

```diff
@@ -195,7 +201,18 @@ func mergeExemplarQueryResponses(results []*ingester_client.ExemplarQueryRespons
 
 	result := make([]mimirpb.TimeSeries, len(exemplarResults))
 	for i, k := range keys {
-		result[i] = exemplarResults[k]
+		ts := exemplarResults[k]
+		for i, l := range ts.Labels {
+			ts.Labels[i].Name = strings.Clone(l.Name)
```

Comment:
Not for this PR, but for all the places where we do cloning of strings like this for `LabelAdapter` instances and then later convert to `labels.Labels`, we should consider just converting to `labels.Labels` instances here. This would save us cloning the strings and then making another copy when we convert to `labels.Labels`.